### PR TITLE
Don't undef rewrite if unused in the ctrl flow

### PR DESCRIPTION
**Pull request by aqjune**
This patch adds `unused_vars` to value analysis, to support the optimization discussed yesterday

### CODE REVIEW

**Comment by aqjune** [comment]
Running LLVM unit tests

File: ir/state.cpp [hunk 1]
```diff
@@ -149,6 +151,16 @@ expr State::strip_undef_and_add_ub(const Value &val, const expr &e) {
            is_undef_cond(not_undef, var);
   };
 
+  auto is_if_undef_or_add = [&](const expr &e, expr &var, expr &not_undef) {
```
Comment (nunoplopes) [inline ir/state.cpp:154]: this is unrelated to the main patch. Can you split this and benchmark the those patches independently pls? Otherwise we have no idea what's causing what.

**Comment by aqjune** [comment]
Split the strip_undef optimization as a separate PR

**Comment by nunoplopes** [comment]
thanks! the patch LGTM. Please let us know the perf results once finished!

**Comment by aqjune** [comment]
Timeout: 4158 -> 4157 ... :/ 
Maybe reestimate its impact with C benchmarks after things are prepared?

**Comment by nunoplopes** [comment]
I'll run on the bechmark suite I've here. Takes a few hours.

**Comment by nunoplopes** [comment]
I see a decent increase in number of UNSAT & trivial queries. This is good, it verifies more stuff, even if # of times out decreases less dramatically.

**Comment by aqjune** [comment]
I'm glad to hear that; thank you so much!
